src/rest-server/src/models/token.js

```diff
@@ -49,58 +65,83 @@ const sign = async (username, application, expiration) => {
  * @param {Object} data - id -> token format data object (data stored in k8s secret)
  * @returns {Object} Purged data in the same format
  */
-const purge = (data) => {
+const purge = async (data, jobSpecific) => {
   const result = {};
   for (const [key, val] of Object.entries(data)) {
     try {
+      // expired tokens will be removed
       jwt.verify(val, secret);
-      result[key] = val;
     } catch (err) {
-      // pass;
+      continue;
+    }
+    if (jobSpecific) {
+      // get framework from db
+      const frameworkName = deserialize(key);
+      const framework = await databaseModel.Framework.findOne({
+        attributes: ['state'],
+        where: { name: encodeName(frameworkName) },
+      });
+      // checkframework state, job-tokens of finished jobs will be removed
+      if (
+        !framework ||
+        ['SUCCEEDED', 'STOPPED', 'FAILED'].indexOf(framework.state) !== -1
```

Comment:
Suggest to use `framework.subState`.

`framework.subState === "Completed"` means the job is finished.

Reply:
> Suggest to use `framework.subState`.
> 
> `framework.subState === "Completed"` means the job is finished.

Fixed, thk